src/halmos/cheatcodes.py

```diff
@@ -230,6 +231,45 @@ def symbolic_storage(ex, arg, sevm, stack, step_id):
     return ByteVec()  # empty return data
 
 
+def snapshot_storage(ex, arg, sevm, stack, step_id):
+    account = uint160(arg.get_word(4))
+    account_alias = sevm.resolve_address_alias(
+        ex, account, stack, step_id, allow_branching=False
+    )
+
+    if account_alias is None:
+        error_msg = f"snapshotStorage() is not allowed for a nonexistent account: {hexify(account)}"
+        raise HalmosException(error_msg)
+
+    return ByteVec(ex.storage[account_alias].digest())
+
+
+def snapshot_state(ex, arg, sevm, stack, step_id):
+    """
+    Generates a snapshot ID by hashing the current state (code, storage, and balance).
+    """
+    m = hashlib.sha3_256()
```

Comment:
thanks for suggestion!

Reply:
wdyt? https://github.com/a16z/halmos/pull/427/commits/7d08e7abd66c83918a5c5926ee939d4d39ca681e

Reply:
for consistency I think it would be better to avoid xxh3_128 and use xxh3_64 only. Plus that way we have an extra 64 bit reserved for some future use!

Reply:
I was considering 128 bits for storages to be more conservative. (Storages are more frequently updated than code and balances.) With 64 bits, the hash collision probability exceeds 10^-12 when generating [6,000 hashes](https://en.wikipedia.org/wiki/Birthday_problem#Probability_table) (which is large, but not unrealistic when running halmos in cloud several days.) With 128 bits, however, it would take 20 trillion hashes to reach the 10^-12 collision probability.

That said, 64 bits might still be good enough for our use case. But, in that case, I'm not sure what's the benefit of using xxhash vs the builtin `hash()`. The builtin hash() can take arbitrary objects as input, and also doesn't seem slower than xxhash:
```
$ python -m timeit --setup="from xxhash import xxh3_64_digest" "xxh3_64_digest(b'')"
5000000 loops, best of 5: 54.2 nsec per loop

$ python -m timeit "hash(b'')"
20000000 loops, best of 5: 12.6 nsec per loop
```
The only concern with the builtin hash() is that it returns 32-bit hashes on 32-bit machines, but I guess 32-bit machines are rare these days, so it might not be a real concern? What do you think?

Reply:
let me merge this as is, and we can discuss hash functions separately.